### PR TITLE
add disableCanonical prop

### PR DESCRIPTION
**Pull request by emielphntm**
Adding disableCanonical so that canonical linking happens when path is empty - needed for the homepages

### CODE REVIEW

File: src/components/MetaHeadEmbed.tsx [hunk 1]
```diff
@@ -66,8 +66,14 @@ export interface MetaEmbedProps {
...
 
+  /**
+   * Enable canonical even if the canonicalUrl is empty.
+   * Mainly used for homepages, where the canonicalPath is empty.
+   */
+  disableCanonical?: boolean;
+
   /** Canonical URL of your webpage that will be used as its default app URL. */
-  canonicalUrl?: string;
+  canonicalPath?: string;
```
Comment (PauloMFJ) [inline src/components/MetaHeadEmbed.tsx:76]: Wondering if this is needed? Can probably just use `pagePath`?

**Review by PauloMFJ — CHANGES_REQUESTED** [review]
Can you

File: src/components/MetaHeadEmbed.tsx [hunk 2]
```diff
@@ -66,8 +66,14 @@ export interface MetaEmbedProps {
   /** Webpage description. Should be less than 160 characters. */
   description: string;
 
+  /**
```
[the remark this answers is not in the document]
Comment (emielphntm) [reply src/components/MetaHeadEmbed.tsx:69]: Done

File: src/components/MetaHeadEmbed.tsx (hunk 1 above)
Comment (emielphntm) [reply src/components/MetaHeadEmbed.tsx]: Removed and added pageUrl taking the pagePath

File: src/components/MetaHeadEmbed.tsx [hunk 3]
```diff
@@ -136,13 +139,11 @@ const MetaHeadEmbed = ({
...
-    (isAbsoluteUrl(canonicalUrl)
-      ? canonicalUrl
-      : `${baseSiteUrl}/${canonicalUrl}`);
-
-  const pageUrl = pagePath ? `${baseSiteUrl}/${pagePath}` : baseSiteUrl;
+  const pathUrl = pagePath
+    ? isAbsoluteUrl(pagePath)
+      ? pagePath
+      : `${baseSiteUrl}/${pagePath}`
+    : `${baseSiteUrl}/${pagePath}`;
```
Comment (PauloMFJ) [inline src/components/MetaHeadEmbed.tsx:146]:
```suggestion
    : baseSiteUrl`;
```

File: src/components/MetaHeadEmbed.tsx [hunk 4]
```diff
@@ -158,10 +159,13 @@ const MetaHeadEmbed = ({
...
     ),
-    canonicalUrl && <link key="canonical" rel="canonical" href={canonical} />,
+
+    !disableCanonical && (
+      <link key="canonical" rel="canonical" href={pathUrl} />
+    ),
 
     <meta key="og:type" property="og:type" content="website" />,
-    <meta key="og:url" property="og:url" content={pageUrl} />,
+    <meta key="og:url" property="og:url" content={pagePath} />,
```
Comment (PauloMFJ) [inline src/components/MetaHeadEmbed.tsx:168]:
```suggestion
    <meta key="og:url" property="og:url" content={pageUrl} />,
```

**Review by PauloMFJ — CHANGES_REQUESTED** [review]

File: src/components/MetaHeadEmbed.tsx (hunk 4 above)
Comment (emielphntm) [reply src/components/MetaHeadEmbed.tsx:167]: 👍

File: src/components/MetaHeadEmbed.tsx (hunk 3 above)
Comment (emielphntm) [reply src/components/MetaHeadEmbed.tsx]: done

File: README.md [hunk 5]
```diff
@@ -61,23 +61,23 @@ export default PageLayout;
...
+| **baseSiteUrl**      | string               | **Yes**  | Base site URL, excluding trailing slash.                                                                                                                                                                                                                                                                               |
+| **pagePath**         | string               | **No**   | The path of the current page, excluding leading slash.                                                                                                                                                                                                                                                                 |
+| **disableCanonical** | string               | **No**   | Disable canonical if not desired. Default to false.                                                                                                                                                                                                                                                                    |
```
Comment (PauloMFJ) [inline README.md:73]:
```suggestion
| **disableCanonical** | boolean               | **No**   | Disable canonical if not desired, defaults to `false`.                                                                                                                                                                                                                                                                    |
```

File: src/components/MetaHeadEmbed.tsx [hunk 6]
```diff
@@ -66,8 +66,10 @@ export interface MetaEmbedProps {
   /** Webpage description. Should be less than 160 characters. */
   description: string;
 
-  /** Canonical URL of your webpage that will be used as its default app URL. */
-  canonicalUrl?: string;
+  /**
+   * Disable canonical if not desired. Default to false.
```
Comment (PauloMFJ) [inline src/components/MetaHeadEmbed.tsx:70]:
```suggestion
   * Disable canonical if not desired, defaults to `false`.
```

File: src/components/MetaHeadEmbed.tsx (hunk 4 above)
Comment (PauloMFJ) [reply src/components/MetaHeadEmbed.tsx:167]: It's still `pagePath`? When it should be `pageUrl`?

**Review by PauloMFJ — CHANGES_REQUESTED** [review]

File: README.md [hunk 7]
```diff
@@ -61,23 +61,23 @@ export default PageLayout;
...
+| **baseSiteUrl**      | string               | **Yes**  | Base site URL, excluding trailing slash.                                                                                                                                                                                                                                                                               |
+| **pagePath**         | string               | **No**   | The path of the current page, excluding leading slash.                                                                                                                                                                                                                                                                 |
+| **disableCanonical** | string               | **No**   | Disable canonical if not desired, defaults to to false.                                                                                                                                                                                                                                                                |
```
Comment (PauloMFJ) [inline README.md:73]:
```suggestion
| **disableCanonical** | boolean               | **No**   | Disable canonical if not desired, defaults to to `false`.                                                                                                                                                                                                                                                                |
```
String needs to be boolean, and should use '`' as its a variable.

File: src/components/MetaHeadEmbed.tsx [hunk 8]
```diff
@@ -66,8 +66,10 @@ export interface MetaEmbedProps {
   /** Webpage description. Should be less than 160 characters. */
   description: string;
 
-  /** Canonical URL of your webpage that will be used as its default app URL. */
-  canonicalUrl?: string;
+  /**
+   * Disable canonical if not desired, defaults to false.
```
Comment (PauloMFJ) [inline src/components/MetaHeadEmbed.tsx:70]:
```suggestion
   * Disable canonical if not desired, defaults to `false`.
```
False is a state so should be surrounded with '`'.

**Review by PauloMFJ — CHANGES_REQUESTED** [review]

File: README.md [hunk 9]
```diff
@@ -61,23 +61,23 @@ export default PageLayout;
...
+| **baseSiteUrl**      | string               | **Yes**  | Base site URL, excluding trailing slash.                                                                                                                                                                                                                                                                               |
+| **pagePath**         | string               | **No**   | The path of the current page, excluding leading slash.                                                                                                                                                                                                                                                                 |
+| **disableCanonical** | string               | **No**   | Disable canonical if not desired, defaults to to `false`.                                                                                                                                                                                                                                                              |
```
Comment (PauloMFJ) [inline README.md:73]: This is wrong. It's a boolean not a string. You can apply my suggestion without having to re-commit btw
```suggestion
| **disableCanonical** | boolean               | **No**   | Disable canonical if not desired, defaults to to `false`.                                                                                                                                                                                                                                                              |
```

**Review by PauloMFJ — CHANGES_REQUESTED** [review]

**Review by PauloMFJ — APPROVED** [review]

File: README.md [hunk 10]
```diff
@@ -61,23 +61,23 @@ export default PageLayout;
...
+| **baseSiteUrl**      | string               | **Yes**  | Base site URL, excluding trailing slash.                                                                                                                                                                                                                                                                               |
+| **pagePath**         | string               | **No**   | The path of the current page, excluding leading slash.                                                                                                                                                                                                                                                                 |
+| **disableCanonical** | boolean              | **No**   | Disable canonical if not desired, defaults to to `false`.                                                                                                                                                                                                                                                              |
```
Comment (PauloMFJ) [inline README.md:73]:
```suggestion
| **disableCanonical** | boolean              | **No**   | Disable canonical if not desired, defaults to `false`.                                                                                                                                                                                                                                                              |
```

**Review by PauloMFJ — CHANGES_REQUESTED** [review]

**Pull request merged**